src/util.rs

```diff
@@ -145,13 +145,24 @@ pub async fn hash_all_archives(
         let path = path.clone();
         pool.spawn_blocking(move || -> Result<String> {
             let mut m = Digest::new();
-            let reader = File::open(&path)
+            let archive_file = File::open(&path)
                 .with_context(|| format!("Failed to open file for hashing: {:?}", path))?;
-            let mut archive = Archive::new(reader);
-            while let Some(entry) = archive.next_entry() {
-                let entry = entry?;
-                m.update(entry.header().identifier());
-                update_from_reader(&mut m, entry)?;
+            let archive_mmap =
+                unsafe { memmap2::MmapOptions::new().map_copy_read_only(&archive_file)? };
```

Comment:
would it be possible to avoid the unsafe ?

Reply:
I don't believe there is, no.

The problem, as I understand it, is that the CoW guarantees are very weak, so its technically possible for someone to mutate a file on disk and have that reflected in map, which then means the `&[u8]` may change, which is unsound. As a result, these APIs are `unsafe`, and users must not mutate files as they're being processed.

Reply:
There is no way around this to my knowledge, but we are talking about an archive that was just created by the compiler and requires access to the cache dir. Hence the user could likely also access the source tree, which is more severe. Hence causing a invalid cache entry is mostly an annoyance in the current trusted setup and unlikely to happen.

Reply:
ok, thanks :)